[v9.5.*] Cannot find name useI18n

### Environment

(From the Stackblitz reproduction repo)
- Operating System: Linux
- Node Version:     v18.20.3
- Nuxt Version:     3.16.2
- CLI Version:      3.24.1
- Nitro Version:    2.11.8
- Package Manager:  pnpm@8.15.6
- Builder:          -
- User Config:      compatibilityDate, devtools, modules, i18n
- Runtime Modules:  @nuxtjs/i18n@9.5.1
- Build Modules:    -

### Reproduction

https://stackblitz.com/edit/github-jjpn8kps
Run `pnpm typecheck` in the terminal.

### Describe the bug

After update to v9.5.* typecheck fails with an error:
```sh
error TS2304: Cannot find name 'useI18n'.

2 const { locales } = useI18n();
                      ~~~~~~~

Found 1 error in components/LangSwitcher.vue:2


ERROR  Process exited with non-zero status (2)                                      ...
```

### Additional context

_No response_

### Logs

```shell
❯ pnpm typecheck

> nuxt-app@ typecheck /home/projects/github-jjpn8kps
> nuxi typecheck


 WARN  [nuxt-i18n]: Unable to import oxc-parser, falling back to @oxc-parser/wasm.   11:10:55 AM


 WARN  Slow module @nuxt/devtools took 15917.19ms to setup.                          11:11:11 AM

components/LangSwitcher.vue:2:21 - error TS2304: Cannot find name 'useI18n'.

2 const { locales } = useI18n();
                      ~~~~~~~


Found 1 error in components/LangSwitcher.vue:2


 ERROR  Process exited with non-zero status (2)                                      11:11:42 AM

    at new x (node_modules/.pnpm/tinyexec@1.0.1/node_modules/tinyexec/dist/main.js:440:5)
    at R._waitForOutput (node_modules/.pnpm/tinyexec@1.0.1/node_modules/tinyexec/dist/main.js:548:13)
    at async Object.run (node_modules/.pnpm/@nuxt+cli@3.24.1/node_modules/@nuxt/cli/dist/chunks/typecheck.mjs:61:7)
    at async runCommand (node_modules/.pnpm/citty@0.1.6/node_modules/citty/dist/index.mjs:337:16)
    at async runCommand (node_modules/.pnpm/citty@0.1.6/node_modules/citty/dist/index.mjs:328:11)
    at async Module.runMain (node_modules/.pnpm/citty@0.1.6/node_modules/citty/dist/index.mjs:466:7) 



 ERROR  Process exited with non-zero status (2)                                      11:11:42 AM

 ELIFECYCLE  Command failed with exit code 1.
```

--- a/src/alias.ts
+++ b/src/alias.ts
@@ -19,10 +19,7 @@ const debug = createDebug('@nuxtjs/i18n:alias')
 
 export function setupAlias({ userOptions: options, isDev, isPrepare }: I18nNuxtContext, nuxt: Nuxt) {
   const modules = {
-    [VUE_I18N_PKG]:
-      isDev || isPrepare
-        ? `${VUE_I18N_PKG}/dist/vue-i18n.mjs`
-        : `${VUE_I18N_PKG}/dist/vue-i18n${options.bundle?.runtimeOnly ? '.runtime' : ''}.mjs`,
+    [VUE_I18N_PKG]: `${VUE_I18N_PKG}/dist/vue-i18n${!isDev && !isPrepare && options.bundle?.runtimeOnly ? '.runtime' : ''}.mjs`,
     [SHARED_PKG]: `${SHARED_PKG}/dist/shared.mjs`,
     [MESSAGE_COMPILER_PKG]: `${MESSAGE_COMPILER_PKG}/dist/message-compiler.mjs`,
     [CORE_BASE_PKG]: `${CORE_BASE_PKG}/dist/core-base.mjs`,

--- a/src/module.ts
+++ b/src/module.ts
@@ -44,6 +44,11 @@ export default defineNuxtModule<NuxtI18nOptions>({
     prepareOptions(ctx, nuxt)
 
     /**
+     * auto imports
+     */
+    prepareAutoImports(ctx)
+
+    /**
      * allow other modules to setup first in case these use i18n hooks
      */
     nuxt.hook('modules:done', async () => {
@@ -100,11 +105,6 @@ export default defineNuxtModule<NuxtI18nOptions>({
        * setup nitro
        */
       await setupNitro(ctx, nuxt)
-
-      /**
-       * auto imports
-       */
-      prepareAutoImports(ctx, nuxt)
 
       /**
        * transpile @nuxtjs/i18n

--- a/src/prepare/auto-imports.ts
+++ b/src/prepare/auto-imports.ts
@@ -1,18 +1,14 @@
-import type { Nuxt } from '@nuxt/schema'
 import {
   NUXT_I18N_COMPOSABLE_DEFINE_CONFIG,
   NUXT_I18N_COMPOSABLE_DEFINE_LOCALE,
   NUXT_I18N_COMPOSABLE_DEFINE_ROUTE,
   VUE_I18N_PKG
 } from '../constants'
-import { addComponent, addImports } from '@nuxt/kit'
+import { addComponent, addImports, resolveModule } from '@nuxt/kit'
 import { runtimeDir } from '../dirs'
 import type { I18nNuxtContext } from '../context'
 
-export function prepareAutoImports({ debug, resolver }: I18nNuxtContext, nuxt: Nuxt) {
-  const vueI18nPath = nuxt.options.alias[VUE_I18N_PKG]
-  debug('vueI18nPath for auto-import', vueI18nPath)
-
+export function prepareAutoImports({ resolver, userOptions: options, isDev, isPrepare }: I18nNuxtContext) {
   addComponent({
     name: 'NuxtLinkLocale',
     filePath: resolver.resolve(runtimeDir, 'components/NuxtLinkLocale')
@@ -23,8 +19,9 @@ export function prepareAutoImports({ debug, resolver }: I18nNuxtContext, nuxt: N
     filePath: resolver.resolve(runtimeDir, 'components/SwitchLocalePathLink')
   })
 
+  const vueI18nPath = `${VUE_I18N_PKG}/dist/vue-i18n${!isDev && !isPrepare && options.bundle?.runtimeOnly ? '.runtime' : ''}.mjs`
   addImports([
-    { name: 'useI18n', from: vueI18nPath },
+    { name: 'useI18n', from: resolveModule(vueI18nPath) },
     ...[
       'useRouteBaseName',
       'useLocalePath',

